'ninja flash' does not work for IMXRT1052 target

**Describe the bug**
Follow the guide to run 'ninja flash', all build processes are passed just stop the last step. It shows 'FAILED: zephyr/cmake/flash/CMakeFiles/flash'

**To Reproduce**
Steps to reproduce the behavior:
1. Install Jlink OpenSDA firmware for the on-board debugger
2. The debugger itself works well, e.g. using J-Link Lite to program a zephyr.bin binary is good.
3. Go to samples/hello_world

```
mkdir build & cd build
cmake -GNinja -DBOARD=mimxrt1050_evk
ninja flash
```

Full error message

```
Using runner: jlink
Flashing Target Device
Traceback (most recent call last):
  File "c:\users\nxa07532\appdata\local\programs\python\python37-32\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "c:\users\nxa07532\appdata\local\programs\python\python37-32\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Users\nxa07532\AppData\Local\Programs\Python\Python37-32\Scripts\west.exe\__main__.py", line 9, in <module>
  File "c:\users\nxa07532\appdata\local\programs\python\python37-32\lib\site-packages\west\_bootstrap\main.py", line 499, in main
    wrap(wrap_argv)
  File "c:\users\nxa07532\appdata\local\programs\python\python37-32\lib\site-packages\west\_bootstrap\main.py", line 485, in wrap
    west.main.main(argv)
  File "C:\Users\nxa07532\zephyrproject\.west\west\src\west\main.py", line 580, in main
    args.handler(args, unknown)
  File "C:\Users\nxa07532\zephyrproject\.west\west\src\west\main.py", line 332, in ext_command_handler
    command.run(*west_parser.parse_known_args(argv))
  File "C:\Users\nxa07532\zephyrproject\.west\west\src\west\commands\command.py", line 90, in run
    self.do_run(args, unknown)
  File "c:\users\nxa07532\zephyrproject\zephyr\scripts\west_commands\flash.py", line 32, in do_run
    'ZEPHYR_BOARD_FLASH_RUNNER')
  File "C:\Users\nxa07532\zephyrproject\zephyr\scripts/west_commands\run_common.py", line 228, in do_run_common
    runner.run(command_name)
  File "C:\Users\nxa07532\zephyrproject\zephyr\scripts/west_commands\runners\core.py", line 407, in run
    self.do_run(command, **kwargs)
  File "C:\Users\nxa07532\zephyrproject\zephyr\scripts/west_commands\runners\jlink.py", line 99, in do_run
    self.flash(**kwargs)
  File "C:\Users\nxa07532\zephyrproject\zephyr\scripts/west_commands\runners\jlink.py", line 150, in flash
    self.check_call(cmd)
  File "C:\Users\nxa07532\zephyrproject\zephyr\scripts/west_commands\runners\core.py", line 466, in check_call
    subprocess.check_call(cmd)
  File "c:\users\nxa07532\appdata\local\programs\python\python37-32\lib\subprocess.py", line 342, in check_call
    retcode = call(*popenargs, **kwargs)
  File "c:\users\nxa07532\appdata\local\programs\python\python37-32\lib\subprocess.py", line 323, in call
    with Popen(*popenargs, **kwargs) as p:
  File "c:\users\nxa07532\appdata\local\programs\python\python37-32\lib\subprocess.py", line 775, in __init__
    restore_signals, start_new_session)
  File "c:\users\nxa07532\appdata\local\programs\python\python37-32\lib\subprocess.py", line 1178, in _execute_child
    startupinfo)
**FileNotFoundError: [WinError 2] The system cannot find the file specified
FAILED: zephyr/cmake/flash/CMakeFiles/flash**
cmd.exe /C "cd /D C:\Users\nxa07532\app\build && "C:\Program Files\CMake\bin\cmake.exe" -E env C:/Users/nxa07532/AppData/Local/Programs/Python/Python37-32/Scripts/west.exe flash --skip-rebuild"
ninja: build stopped: subcommand failed.
```

--- a/scripts/west_commands/runners/jlink.py
+++ b/scripts/west_commands/runners/jlink.py
@@ -7,6 +7,7 @@
 import os
 import tempfile
 import sys
+import shutil
 
 from west import log
 from runners.core import ZephyrBinaryRunner, RunnerCaps, \
@@ -33,7 +34,7 @@ class JLinkBinaryRunner(ZephyrBinaryRunner):
         self.commander = commander
         self.flash_addr = flash_addr
         self.erase = erase
-        self.gdbserver_cmd = [gdbserver]
+        self.gdbserver = gdbserver
         self.iface = iface
         self.speed = speed
         self.gdb_port = gdb_port
@@ -85,8 +86,12 @@ class JLinkBinaryRunner(ZephyrBinaryRunner):
     def print_gdbserver_message(self):
         log.inf('J-Link GDB server running on port {}'.format(self.gdb_port))
 
+    def check_cmd(self, cmd):
+        if shutil.which(cmd) is None:
+            log.die('{} is not installed or cannot be found'.format(cmd))
+
     def do_run(self, command, **kwargs):
-        server_cmd = (self.gdbserver_cmd +
+        server_cmd = ([self.gdbserver] +
                       ['-select', 'usb', # only USB connections supported
                        '-port', str(self.gdb_port),
                        '-if', self.iface,
@@ -98,9 +103,11 @@ class JLinkBinaryRunner(ZephyrBinaryRunner):
         if command == 'flash':
             self.flash(**kwargs)
         elif command == 'debugserver':
+            self.check_cmd(self.gdbserver)
             self.print_gdbserver_message()
             self.check_call(server_cmd)
         else:
+            self.check_cmd(self.gdbserver)
             if self.gdb_cmd is None:
                 raise ValueError('Cannot debug; gdb is missing')
             if self.elf_name is None:
@@ -117,6 +124,7 @@ class JLinkBinaryRunner(ZephyrBinaryRunner):
             self.run_server_and_client(server_cmd, client_cmd)
 
     def flash(self, **kwargs):
+        self.check_cmd(self.commander)
         if self.bin_name is None:
             raise ValueError('Cannot flash; bin_name is missing')
 

